src/connector.ts

```diff
@@ -1,64 +1,127 @@
-import { Signer } from 'ethers'
-import { ZeroWalletConnectorOptions, ZeroWalletSigner } from './types'
+import { ethers, Signer } from 'ethers'
+import { ZeroWalletConnectorOptions } from './types'
 import { ZeroWalletProvider } from './provider'
+import { ZeroWalletSigner } from './signer'
 import { Chain, Connector, ConnectorData } from 'wagmi'
+import { StorageFactory } from 'store/storageFactory'
+import { IStoreable } from 'store/IStoreable'
 
 export class ZeroWalletConnector extends Connector<ZeroWalletProvider, ZeroWalletConnectorOptions, ZeroWalletSigner> {
-	readonly id = 'zero-wallet'
-	readonly name = 'Zero Wallet'
+    readonly id = 'zero-wallet'
+    readonly name = 'Zero Wallet'
 
-	private provider: ZeroWalletProvider
+    private provider: ZeroWalletProvider
+    private store: IStoreable
 
-	constructor(config: { chains?: Chain[]; options: ZeroWalletConnectorOptions }) {
-		super(config)
-		this.provider = new ZeroWalletProvider(config.options.jsonRpcProviderUrl)
-	}
+    constructor(config: { chains?: Chain[]; options: ZeroWalletConnectorOptions }) {
+        super(config)
+        this.store = StorageFactory.create(config.options.store);
+        const _chain = 
+            (config?.chains && config.chains.length > 0) ? 
+            { chainId: config.chains[0].id, name: config.chains[0].name } :
+            { chainId: 1, name: 'Ethereum' }
 
-	get ready() {
-		return true
-	}
+        this.provider = new ZeroWalletProvider(config.options.jsonRpcProviderUrl, _chain, this.store, config.options.recoveryMechanism)
+    }
 
-	async connect(): Promise<Required<ConnectorData>> {
-		throw new Error('not implemented')
-	}
+    get ready() {
```

Comment:
isReady()

Reply:
This is the template of the wagmi connector. @adiwajshing suggested we follow this template.